### PR TITLE
add conformance test when applying null with operators

### PR DESCRIPTION
**Pull request by xd1313113**
Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-tests-data/eval/primitives/operators/and.ion [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  
+  // AND with MISSING tests
+  {
+    name: "true AND MISSING",
+    statement: "true AND MISSING",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [EvalModeCoerce, EvalModeError],
+      output: $missing::null
+    }
```
Comment (xd1313113) [inline partiql-tests-data/eval/primitives/operators/and.ion:95]: Currrently true and missing is returning null instead of $missing::null

File: partiql-tests-data/eval/primitives/operators/or-operator.ion [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    name: "MISSING OR true",
+    statement: "MISSING OR true",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [EvalModeCoerce, EvalModeError],
+      output: true
+    }
+  },
+  {
+    name: "false OR MISSING",
```
Comment (xd1313113) [inline partiql-tests-data/eval/primitives/operators/or-operator.ion:107]: false or missing return null instead of $missing::null

**Comment by xd1313113** [comment]
> For date_ion, I am not sure how we want to enable it

**Comment by XuechunHHH** [comment]
We can probably include some `NULL/MISSING` tests with `*` and `/` operator as well in this PR.

File: partiql-tests-data/eval/primitives/operators/minus.ion [hunk 3]
```diff
@@ -309,5 +309,205 @@ minus::[
...
+  },
+  {
+    name: "MISSING - INTERVAL '3' DAY",
+    statement: "MISSING - INTERVAL '3' DAY",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [EvalModeCoerce, EvalModeError],
+      output: $missing::null
+    }
   }
```
Comment (XuechunHHH) [inline partiql-tests-data/eval/primitives/operators/minus.ion:512]: We can also include some tests on:

- `INTERVAL` with range e.g. `INTERVAL '1-1' YEAR TO MONTH` and `INTERVAL '1 01:01:10.123' DAY TO SECOND`.
- `TIMESTAMP` and `TIME` with time zone.

**Review by alancai98 — COMMENTED** [review]
Initial set of tests look correct. Let's add some more per Xuechun + my suggestions

**Comment by alancai98** [comment]
PR is in draft any reason it's still in draft?

**Comment by xd1313113** [comment]
> PR is in draft any reason it's still in draft?

Was pending a question and forget to publish the PR
